metadata-ingestion/src/datahub/ingestion/source/schema_inference/base.py

```diff
@@ -0,0 +1,17 @@
+from io import TextIOWrapper
+from typing import List
+
+from datahub.metadata.com.linkedin.pegasus2avro.schema import SchemaField
+
+
+class SchemaInferenceBase:
+    """
+    Base class for file schema inference.
+    """
+
+    @staticmethod
```

Comment:
Can we get rid of the `@staticmethod` here make it a proper instance method? The reason is if one of the derived classes has additional state maintained at instance level(desired), it cannot be accessed from a static method.